File: NextSpring/src/main/kotlin/com/nextchess/data/OpeningDTO.kt
package com.nextchess.data

import com.nextchess.entities.Variation

open class OpeningDTO(
    val id: Long,
    val name: String,
    val description: String? = null,
    val baseMovesSequence: List<String>,
    val variations: List<Variation>,
    val difficulty: String?
)

File: next-chess/src/models/OpeningDTO.ts
interface OpeningDTO {
    id: string | number;
    name: string;
    description: string;
    baseMovesSequence: string;
    variations: VariationDTO[];
    difficulty: string | number;
}

interface DetailedOpeningDTO extends OpeningDTO {
    masterGames: MasterGameDTO[];
}
